create V1 of article guardian lambda versions (#1)
PChol22/kumo-articles: blog-posts/guardian/lambda-versions/article.md

---
published: false

> title: 'AWS Lambda Versions : Time to clean up!'

Reviewer: No space before colon.

Reviewer 2: Ensure you have coherence between articles of the same serie.
For exemple, for Lift's articles, all titles ended with `Designing Lift`

What do you think of `AWS Lambda Versions : Time to clean up! - Guardian is watching over you` and also change @ealain article title to `AWS Lambda 101: Shave That Bundle Down - Guardian is watching over you`

Reviewer: okpm

> cover_image: https://raw.githubusercontent.com/pchol22/kumo-articles/master/blog-posts/guardian/lambda-versions/assets/cover_image.png

Reviewer: Top!

description: ''
tags: serverless, lambda, quality, AWS
series:
canonical_url:
---

_This article is part of a series on [Guardian][guardian], an open-source, highly configurable, automated best-practice audit tool for AWS serverless architectures._

# Let's keep track

Lambda versioning is great and might come in handy, but do you really need to keep dozens of outdated copies of your functions' code ? Here are two reasons why you should get rid of them.

## 1. Less is more

> It sure can help to have a backup version of your code to rollback to in case of an emergency, but the day it happens, you will feel a lot more conformable only having to chose between two or three documented, aliased versions than being overwhelmed by dozens of anonymous and forgotten ones.

Reviewer 3: Tu voulais utiliser 'conformable' ou 'comfortable'?

Author: ah oui en effet c'était comfortable

> ![Button meme](./assets/button_meme.png 'Button Meme')

Reviewer: Nice!

## 2. Beware of AWS Lambda quotas

> AWS Lambda enforces a regional [75GB limit][quotas] for all your uploaded packages. This threshold might seem hard to reach and harmless, but it takes into account every version of every lambda you have deployed: dozens of versions per function multiplied by a large bundle size (see our [previous article][guardian-bundle-size-article]) will take you to past this ceiling in no time.

Reviewer: Space: 75 GB

> AWS Lambda enforces a regional [75GB limit][quotas] for all your uploaded packages. This threshold might seem hard to reach and harmless, but it takes into account every version of every lambda you have deployed: dozens of versions per function multiplied by a large bundle size (see our [previous article][guardian-bundle-size-article]) will take you to past this ceiling in no time.

Reviewer 2: ```suggestion
AWS Lambda enforces a regional [75GB limit][quotas] for all your uploaded packages. This threshold might seem hard to reach and harmless, but it takes into account every version of every Lambda you have deployed: dozens of versions per function multiplied by a large bundle size (see our [previous article][guardian-bundle-size-article]) will take you over this threshold in no time.
```

Lambda always takes a capital `L`

> It is possible to raise this soft limit by contacting the support, but it's not a sustainable workaround, and you don't want to be stuck unable to upload when there's an important fix to deploy.

Reviewer 2: ```suggestion
It is possible to raise this soft limit by contacting the support, but it's not a sustainable workaround, and you don't want to be stuck, unable to upload your new version, when there's an important fix to deploy.
```

# Easily monitor your lambdas versioning with Guardian 🛡️

[Guardian][guardian] now offers a **new rule** preventing your lambdas versions counter from going crazy.

> ![Rule in CI](./assets/rule_CI.png 'Rule in CI')

Reviewer: I think this screenshot does not put Guardian forward. Do you have the same with all green?

Author: that's true letme check

Guardian also comes with **many other rules** to help you make the best decisions for your Serverless project. It will help you identify where your deployed resources can be optimized to achieve better performance at a lower cost.

# Guardian how-to

```
npm install @kumo-by-theodo/guardian
npx guardian -p <your_aws_profile> -c <your_stack_name>
```

Guardian is available on [NPM][npm-registry]. You will find instructions to use Guardian in your CI.

# See also

> There also many other tools helping you manage your lambdas versioning. One of them is [Serverless Prune Plugin][serverless-prune-plugin], which integrates with the Serverless Framework and allows you to only keep a certain amount of recent versions for each one of your functions.

Reviewer 2: ```suggestion
There also many other tools helping you manage your lambdas versioning. One of them is [Serverless Prune Plugin][serverless-prune-plugin], which integrates with the Serverless Framework and allows you to only keep a certain amount of recent versions for each of your functions.
```
